testing/src/test/java/org/motechproject/nms/testing/it/ma/MobileAcademyServiceBundleIT.java

```diff
@@ -198,8 +205,30 @@ public void testSetLastBookmark() {
     }
 
     @Test
+    public void testCompletionCount() {
+
+        long callingNumber = 9876543210L;
+        int completionCountBefore = activityDataService.findRecordsForUser(String.valueOf(callingNumber)).size();
+        MaBookmark bookmark = new MaBookmark(callingNumber, 666L, null, null);
+        maService.setBookmark(bookmark);
+        List<Bookmark> added = bookmarkDataService.findBookmarksForUser("9876543210");
+        assertTrue(added.size() == 1);
+
+        bookmark.setBookmark("Chapter11_Quiz");
+        Map<String, Integer> scores = new HashMap<>();
+        for (int i = 1; i < 12; i++) {
+            scores.put(String.valueOf(i), ((int) (Math.random() * 100)) % 5);
+        }
+        bookmark.setScoresByChapter(scores);
+        maService.setBookmark(bookmark);
+        int completionCountAfter = activityDataService.findRecordsForUser(String.valueOf(callingNumber)).size();
+
+        assertEquals(completionCountBefore + 1, completionCountAfter);
```

Comment:
Yes, but I would also be assuming that the earlier call to the db succeeded (which it should have).. but in the event that the clear failed, this would still pass?